### ISSUE TITLE
Node is not reloaded after script.py change

### ISSUE THREAD

**Issue by scienceworld** [museumvictoria/nodel#48]
I was upgrading our adamio nodes from the legacy recipe to the new non-modbus recipe.  when I replaced the script.py file in the node folder Nodel did not refresh the script.

I then tried editing the script, pasting in the new code and hitting save.  Nodel saved the file and created a backup, but did not refresh the node to accept the changes.  I have to delete the node and recreate the node to get it to reprocess the script. or rename the node.

![image](https://cloud.githubusercontent.com/assets/3997916/12859426/f2fc1ba0-cc0a-11e5-81aa-bac22a31e469.png)

from the screenshot you can see the last time Main was processed was at 11am.  I made the changes at 1pm.

**Comment by scienceworld** [comment]
So I tried replacing the script via the editor, and it updated on a new node

The issue might be related to _how_ I replaced the script.  

1. deleted all files in the node
2. put in new script.py
3. went to nodel to view the node, and the old script/parameters were there
4. tried replacing the script in the editor and no effect

so nodel might not have noticed the change in file contents and didnt refresh the node anymore.

**Comment by justparking** [comment]
Being able to change the `script.py` externally is fundamental behaviour so I looked through the code and worked out your problem: If `nodeConfig.json` still existed, it would have picked up the direct changes and behaved the way you expected.

The code reveals that *while `nodeConfig.json` exists*, it'll check for updates to `nodeConfig.json` and/or `script.py` every 10 seconds or sooner. Then, if either `nodeConfig.json` or `script.py` files have changed (through /Save or direct file updates), the node recycles itself. It was coded that way to avoid a particular race condition that we came across in alpha versions of the code.

I will post a Wiki article soon explaining the node creation/destruction/recycle logic that Nodel Host's employ.
